assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/data/useDataReducer.ts

```diff
@@ -0,0 +1,88 @@
+import { append, findIndex, insert, propEq, update } from 'ramda';
+
+import { DataStateReducer, StateInitializer, DataState } from './types';
+
+export const initialState: DataState = {
+	ticket: null,
+	prices: [],
+	deletedPrices: [],
+};
+
+const useDataReducer = (initializer: StateInitializer): DataStateReducer => {
+	const dataReducer: DataStateReducer = (state, action) => {
+		const { type, id, index, fieldValues, ticketPrice, price } = action;
+
+		switch (type) {
+			case 'TOGGLE_REVERSE_CALCULATE':
+				return {
+					...state,
+					ticket: {
+						...state.ticket,
+						reverseCalculate: !state.ticket.reverseCalculate,
+					},
+				};
+			case 'UPDATE_TICKET_PRICE':
+				return {
+					...state,
+					ticket: {
+						...state.ticket,
+						price: ticketPrice,
+					},
+				};
+
+			case 'ADD_PRICE':
+				const newPrices =
+					typeof index !== 'undefined' ? insert(index, price, state.prices) : append(price, state.prices);
+				return {
+					...state,
+					prices: newPrices,
+				};
+
+			case 'UPDATE_PRICE':
+				// find the index of the price to update
+				const priceIndex = findIndex(propEq('id', id))(state.prices);
```

Comment:
might also be able to use `entityHasGuid()` from `/domain/shared/services/predicates/selectionById` 

or better yet... 

move the above into that file (modified accordingly of course) and then import here.

Question:

is there any benefit to using the curried version of this:

```js
findIndex(propEq('id', id))(state.prices)
```

vs:

```js
findIndex(propEq('id', id), state.prices)
```

???

Reply:
No extra benefit.

Reply:
Removed the curried version

Reply:
used `entityHasGuid` 👍